# Multiple Linear Regression

### Importing the Libraries

In [12]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

### Importing the dataset

In [14]:
dataset = pd.read_csv('Data.csv')
X = dataset.iloc[:, :-1].values  # This selects all the columns except the last one because that is the dependent variable
y = dataset.iloc[:, -1].values   # This selects only the columns except others because that is the dependent variable

### Splitting the dataset into the training set and Test set

In [25]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 1)

### Training the Mutiple Linear Regression Model on the Training set

In [21]:
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)
# Model Created

LinearRegression()

### Predicting the Test set Result

In [22]:
y_pred = regressor.predict(X_test)
np.set_printoptions(precision = 2) # this will display any numerical value with only two decimals after comma
print(np.concatenate((y_pred.reshape(len(y_pred),1), y_test.reshape(len(y_test),1)), 1))

[[457.26 458.96]
 [466.72 463.29]
 [440.37 435.27]
 ...
 [476.41 476.22]
 [424.62 440.29]
 [463.91 467.92]]


The left values in the above array are the predicted values and the right ones are the actual values of the test set. We can see that those two are close to each other.

### Evaluate the Model Performance

In [24]:
from sklearn.metrics import r2_score
r2_score(y_test, y_pred)

0.9321860060402447

The $R^2$ for this multiple linear regression model is $0.9321860060402447$